Sources/KlaviyoCore/Models/APIModels/CreateEventPayload.swift

```diff
@@ -0,0 +1,134 @@
+
+            public struct Profile: Equatable, Codable {
+                public let data: ProfilePayload
+
+                public init(data: ProfilePayload) {
+                    self.data = data
+                }
+            }
+
+            public let metric: Metric
+            public var properties: AnyCodable
+            public let profile: Profile
+            public let time: Date
+            public let value: Double?
+            public let uniqueId: String
+            public init(name: String,
+                        properties: [String: Any]? = nil,
+                        email: String? = nil,
+                        phoneNumber: String? = nil,
+                        externalId: String? = nil,
+                        anonymousId: String? = nil,
+                        value: Double? = nil,
+                        time: Date? = nil,
+                        uniqueId: String? = nil) {
+                metric = Metric(name: name)
+                self.properties = AnyCodable(properties ?? [:])
+                self.value = value
+                self.time = time ?? environment.date()
+                self.uniqueId = uniqueId ?? environment.uuid().uuidString
+                profile = Profile(
+                    data: ProfilePayload(
+                        email: email,
+                        phoneNumber: phoneNumber,
+                        externalId: externalId,
+                        anonymousId: anonymousId ?? "")
+                )
+            }
+
+            enum CodingKeys: String, CodingKey {
+                case metric
+                case properties
+                case profile
+                case time
+                case value
+                case uniqueId = "unique_id"
+            }
+        }
+
+        var type = "event"
+        public var attributes: Attributes
+        public init(name: String,
+                    properties: [String: Any]? = nil,
+                    email: String? = nil,
+                    phoneNumber: String? = nil,
+                    externalId: String? = nil,
+                    anonymousId: String? = nil,
+                    value: Double? = nil,
+                    time: Date? = nil,
+                    uniqueId: String? = nil,
+                    pushToken: String? = nil) {
+            attributes = Attributes(
+                name: name,
+                properties: properties?.appendMetadataToProperties(pushToken: pushToken),
+                email: email,
+                phoneNumber: phoneNumber,
+                externalId: externalId,
+                anonymousId: anonymousId,
+                value: value,
+                time: time,
+                uniqueId: uniqueId)
+        }
+    }
+
+    public var data: Event
+    public init(data: Event) {
+        self.data = data
+    }
+}
+
+extension Dictionary where Key == String, Value == Any {
+    fileprivate func appendMetadataToProperties(pushToken: String?) -> [String: Any]? {
```

Comment:
Earlier this is explicitly called from `KlaviyoAPI` after the `CreateEventPayload` instance was created. Seemed like it was possible to simply include it _silently_ when the model is initialized with properties that the user supplies with public facing `Event` data model.

Reply:
Ok just fyi might result in some stale data and slightly different behavior. We will essentially be capturing a point in time vs the most up to date device info.

Reply:
Oh I didn't think about it that way at all. That makes a lot of sense - I'll revert this logic.

Reply:
Based on conversation on slack keeping this as is as we want the metadata to reflect the state of the device when the event was enqueued vs. dequeued.